src/Avalonia.Controls/Application.cs

```diff
@@ -155,7 +155,7 @@ public IResourceDictionary Resources
 
         /// <inheritdoc/>
         bool IResourceNode.HasResources => (_resources?.HasResources ?? false) ||
-            (((IResourceNode?)_styles)?.HasResources ?? false);
+            ((_styles as IResourceNode)?.HasResources ?? false);
```

Comment:
Why cast was changed here?

Reply:
It is workaround to suppress  the warning CS8632: The annotation for nullable reference types should only be used in code within a '#nullable'.